src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,55 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+
+        if (user == null) {
+            throw new InvalidDataException("User cannot be null");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("User login cannot be null");
+        }
+        if (Storage.people.contains(storageDao.get(user.getLogin()))) {
```

Comment:
The condition `Storage.people.contains(storageDao.get(user.getLogin()))` might not work as intended if `storageDao.get(user.getLogin())` returns null. Consider checking if the user exists in a more reliable way, such as directly checking if the login exists in the storage.

Reply:
check like this (storageDao.get(user.getLogin()) != null) {